app/Http/Controllers/ProductController.php

```diff
@@ -37,23 +37,56 @@ public function create()
         return view('addproduct');
     }
 
-    public function store(Request $request){
-
-        $product = new Product;
-
-        $request->validate([
-            'productName' => 'required|unique:products|string',
-            'productBrand' => 'nullable|string',
-            'productPictureUrl' => 'nullable|active_url',
-            'added_by' => 'integer',
+    public function store(Request $request)
+    {
+        $product = new Product();
+        $this->validate($request, [
+            'name' => 'required|unique:products|string',
+            'brand' => 'nullable|string',
+            'picture' => 'nullable|active_url',
         ]);
 
-        $product->name = request('productName');
-        $product->brand = request('productBrand');
-        $product->picture = request('productPictureUrl');
+        $product->name = $request->get('name');
+        $product->brand = $request->get('brand');
+        $product->picture = $request->get('picture');
         $product->added_by = Auth::user()->id;
         $product->save();
 
         return redirect('home');
     }
+
+    public function delete()
+    {
+
+    }
+
+    public function edit($id)
+    {
+        $product = Product::findOrFail($id);
+        return view('editproduct')->with([
```

Comment:
can you plz pass the whole product and not separate it to id, name, brand, picture?
```
return view('editproduct')->with([
   'product' => $product
]);
```
I am not sure but I think you can also do this
```
return view('editproduct')->withProduct();
```

Reply:
Why would it be necessary to pass more than the view needs? It can be beneficial to pass the full object, but it's definitely not wrong to just pass what you need.

Reply:
Of course it is not wrong, but in case of a future addition of a field, you will need to touch the controller as well. That was my thought